### PR TITLE
Implement debug logging flag for requests

### PR DESCRIPTION
**Pull request by TheMacies**
## Description of the change

This change allows users to optionally debug their queries and responses from the server.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue);
- [x] New feature (non-breaking change that adds functionality);
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected);
- [ ] Other (miscellaneous, GitHub workflow changes, changes to the PR template);

### Development

- [ ] Lint rules pass locally;
- [ ] All tests related to the changed code pass in development;

### Code review

- [ ] This pull request has a descriptive title and sufficient context for a reviewer. There may be a screenshot or screencast attached;
- [ ] This pull request is no longer a draft;

### Deployment

- [ ] Selected merge strategy is [squash merge](https://docs.github.com/en/github/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-pull-request-commits);
- [ ] Changes have been reviewed by at least one other engineer;

### CODE REVIEW

File: logging/http_client.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+// Do performs an HTTP request and logs the request and response.
+func (cli *HTTPClient) Do(req *http.Request) (*http.Response, error) {
+	interpretControlSequences := func(text string) string {
+		text = strings.ReplaceAll(text, `\n`, "\n")
+		text = strings.ReplaceAll(text, `\t`, "\t")
+		return text
+	}
+
```
Comment (cube2222) [inline logging/http_client.go:28]: Do we want to also print headers?

Comment (cube2222) [reply logging/http_client.go:28]: Could be useful.

**Review by cube2222 — APPROVED** [review]
oen nit

File: logging/http_client.go (hunk 1 above)
Comment (TheMacies) [reply logging/http_client.go:31]: added headers and path + deffer to write all at once to avoid printing races

**Review by cube2222 — APPROVED** [review]

**Pull request merged**